contracts/src/steel/OpSteel.sol

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.9;
+
+import {Encoding, Steel} from "./Steel.sol";
+
+abstract contract OpCommitmentValidator {
+    /// @notice Address of the OptimismPortal2 contract.
+    IOptimismPortal2 public immutable optimismPortal;
+
+    constructor(address _optimismPortal) {
+        optimismPortal = IOptimismPortal2(_optimismPortal);
+    }
+
+    /// @notice Validates if the Steel commitment.
+    /// @return True if the commitment is valid, false otherwise.
+    function validateCommitment(Steel.Commitment memory commitment) internal view returns (bool) {
+        (uint240 blockID, uint16 version) = Encoding.decodeVersionedID(commitment.id);
+        if (version == 0x100) {
```

Comment:
hmm, I like this idea. We could use `sha256("Beacon")` or something like that for the selector. However, I see two complications:
- The version is only a `uint16`, with only two bytes there could be accidental collisions. (However, it might be possible to change this to a `uint32` without breaking anything.)
- Currently we have `0x00`, `0x01` for block and beacon. So I guess it would be a breaking change if we changed this to a selector as well, and using a selector only for other types is also weird.

Reply:
I suppose this will always need to be somewhat manual. I guess this is more akin to "chain IDs" than Solidity function selectors, in the sense that someone picking a new version number (essentially just us for now) will need to be aware of and avoid colliding with the other version numbers that are in use. But this is manageable.